### PR TITLE
DOC: Update citation file with JOSS paper reference

### LINKED ISSUE #495: DOC: update citation guidelines and link to docs
## Description
Update the [citation guidelines](https://github.com/skypyproject/skypy/blob/main/CITATION.rst) to include the [JOSS paper](https://joss.theoj.org/papers/10.21105/joss.03056) and link the `rst` file to the [documentation](https://skypy.readthedocs.io/en/latest/) page.

### PR DESCRIPTION
**Pull request by Lucia-Fonseca**
## Description
This PR updates the [citation guidelines](https://github.com/skypyproject/skypy/blob/main/CITATION.rst) to include the [JOSS paper](https://joss.theoj.org/papers/10.21105/joss.03056) and link the `rst` file to the [documentation](https://skypy.readthedocs.io/en/latest/) page. Merging this PR successfully closes #495 .
## Checklist
- [x] Follow the [Contributor Guidelines](https://github.com/skypyproject/skypy/blob/main/CONTRIBUTING.rst)
- [ ] Write unit tests
- [x] Write documentation strings
- [x] Assign someone from your working team to review this pull request
- [x] Assign someone from the infrastructure team to review this pull request

### CODE REVIEW

**Comment by Lucia-Fonseca** [comment]
This is done. Can anyone check the bib file and the Zenodo reference therein? Thank you!

**Comment by Lucia-Fonseca** [comment]
Docks are failing but it was working fine on my computer

**Comment by rrjbca** [comment]
> Docs are failing but it was working fine on my computer

You should add `developer/citation` under the developer toctree in index.rst. Although I would suggest moving citation.rst into docs/project as this seems more appropriate than docs/developer

**Comment by Lucia-Fonseca** [comment]
The `Download BibTeX` button is not working. Any ideas? It might be spaces but I cannot make it work consistently

File: CITATION.rst
Comment (rrjbca) [inline CITATION.rst:29]:
```suggestion
         :download:`Download BibTeX <../../main.bib>`
```

Comment (Lucia-Fonseca) [reply CITATION.rst:29]: But the `main.bib` and `CITATION.rst` live at the same level of the directory. Is there a better place for the `main.bib` file?

Comment (rrjbca) [reply CITATION.rst:29]: But the docs are building the page `docs/project/citation.rst` so the path is relative to that

**Comment by Lucia-Fonseca** [comment]
Is there a better place for the `main.bib` file?

**Comment by Lucia-Fonseca** [comment]
Ready for review

**Comment by rrjbca** [comment]
I don't think having a bib file and a download link is a good idea:
- The download link doesn't render in GitHub: https://github.com/Lucia-Fonseca/skypy/blob/citation/CITATION.rst
- The zenodo reference necessarily points to a single release. We ask people to cite the version they used. Even if we kept it up to date with the most recent release (we won't) it is the wrong reference for people using older versions.

**Comment by Lucia-Fonseca** [comment]
> I don't think having a bib file and a download link is a good idea:
Oh, I liked it but yes, it does not render in GitHub.
> 
> * The download link doesn't render in GitHub: https://github.com/Lucia-Fonseca/skypy/blob/citation/CITATION.rst
> * The zenodo reference necessarily points to a single release. We ask people to cite the version they used. Even if we kept it up to date with the most recent release (we won't) it is the wrong reference for people using older versions.
True

**Comment by ntessore** [comment]
We could display the JOSS badge somewhere as well: [DOI](https://doi.org/10.21105/joss.03056)

**Comment by rrjbca** [comment]
> We could display the JOSS badge somewhere as well: [DOI](https://doi.org/10.21105/joss.03056)

It's already on the README. Do you want it in the docs also?

**Comment by ntessore** [comment]
Just an idea, I like that big green box. On the other hand, someone on the CITATION page has already clicked and gone there on purpose, so perhaps not necessary to decorate the page any further.

**Comment by rrjbca** [comment]
It could work:

The BibTex entries can be found at the following links:
[DOI](https://doi.org/10.21105/joss.03056) [DOI](https://doi.org/10.5281/zenodo.3755531)

**Review by rrjbca — COMMENTED** [review]
I would suggest asking for a single acknowledgement with two citations rather than one citation in the main text and another in an acknowledgement. It is a simpler and more clearly worded request that I believe users are more likely to comply with. Also `main.bib` contains a BibTex reference to our Zenodo Concept DOI, but we ask users to cite the relevant Version DOI. I don't see how we can practically provide the BibTex keys without knowing what version they are using, so I would get rid of them. My suggestion would be:

```
Citation Guidelines
===================

[![JOSS](https://joss.theoj.org/papers/10.21105/joss.03056/status.svg)](https://doi.org/10.21105/joss.03056) [![Zenodo](https://zenodo.org/badge/doi/10.5281/zenodo.4475347.svg)](https://doi.org/10.5281/zenodo.3755531)

If you use SkyPy for work or research presented in a publication (whether
directly, or as a dependency of another package) we recommend and encourage
the following acknowledgment:

  This research made use of SkyPy, a Python package for forward modeling
  astronomical surveys (Amara et. al., 2021, SkyPy Collaboration, 202x).

where the citations are to our publication in the `Journal of Open Source
Software`_ and and the `Zenodo`_ DOI for the specific version of the software that
you used. We also encourage citations within the main text wherever
appropriate. DOIs and BibTeX keys are available through the links above.

.. _Journal of Open Source Software: https://joss.theoj.org/papers/10.21105/joss.03056
.. _Zenodo: https://zenodo.org/record/3755531
```

**Comment by Lucia-Fonseca** [comment]
The badges
[JOSS](https://doi.org/10.21105/joss.03056) [Zenodo](https://doi.org/10.5281/zenodo.3755531)

do not render well in the docs

File: CITATION.rst
Comment (ntessore) [inline CITATION.rst:4]:
```suggestion
|JOSS| |Zenodo|
```

File: CITATION.rst
Comment (ntessore) [inline CITATION.rst:19]:
```suggestion
.. _Zenodo DOI: https://zenodo.org/record/3755531

.. |Zenodo| image:: https://zenodo.org/badge/221432358.svg
    :target: https://zenodo.org/badge/latestdoi/221432358
    :alt: SkyPy Concept DOI

.. |JOSS| image:: https://joss.theoj.org/papers/d4fac0604318190d6627ab29b568a48d/status.svg
    :target: https://joss.theoj.org/papers/d4fac0604318190d6627ab29b568a48d
```

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst]: Thanks! Exactly what I was doing

**Comment by Lucia-Fonseca** [comment]
Ready

File: CITATION.rst
Comment (itrharrison) [inline CITATION.rst:12]: Apologies if I have missed discussions which already address these points.
- Are we happy with 'Amara et al' or would another 'SkyPy Collaboration' be the preferred style? I appreciate there are probably reasons for this I have missed!
- We could do the work for the user of including the most up to date version year/reference, which I guess would just require changing as part of each new release.

File: CITATION.rst
Comment (rrjbca) [inline CITATION.rst:11]:
```suggestion
.. admonition:: Acknowledgement

  This research made use of SkyPy, a Python package for forward modeling
```

Just an idea; I think this improves the html docs but doesn't display well in GitHub. What do people think?

File: CITATION.rst
Comment (rrjbca) [reply CITATION.rst:12]: > * Are we happy with 'Amara et al' or would another 'SkyPy Collaboration' be the preferred style? I appreciate there are probably reasons for this I have missed!

"SkyPy Collaboration" isn't listed as an author on the JOSS paper, their template wouldn't allow it. Are you suggesting that people cite it as SkyPy Collaboration 2021 regardless? Would that work with ADS etc?

> * We could do the work for the user of including the most up to date version year/reference, which I guess would just require changing as part of each new release.

But then if somebody is using an older version they shouldn't be citing the most up to date version. It might be possible that for each release the published documentation give the corresponding DOI, but only if we can reserve zenodo DOIs before me make the release. As our Zenodo admin could you investigate this?

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst:11]: I like it.

Comment (Lucia-Fonseca) [reply CITATION.rst:11]: I like it

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst:12]: I am not happy with Amara et al. either, if we could do `SkyPy Collaboration` and not mess up with ADS I would go for it.

File: docs/index.rst
Comment (Lucia-Fonseca) [reply docs/index.rst:15]: Your suggestion wouldn't work.
I could do
```suggestion
   please follow our :doc:`citation guidelines <project/citation>`.
```

Comment (rrjbca) [reply docs/index.rst]: I tested it and it does work. It takes the hyperlink text from the heading on the linked page.

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst:14]: I cannot find anything in the ADS FAQ to answer this...

File: docs/index.rst
Comment (Lucia-Fonseca) [reply docs/index.rst]: OK.

File: CITATION.rst
Comment (rrjbca) [reply CITATION.rst:13]: Readthedocs looks good: https://skypy--496.org.readthedocs.build/en/496/project/citation.html
GitHub not so good: https://github.com/Lucia-Fonseca/skypy/blob/citation/CITATION.rst
While I like how the documentation looks and think it is more important, I don't know if people are happy with the ugly rendering on GitHub

File: CITATION.rst
Comment (rrjbca) [reply CITATION.rst:14]: I don't think it is possible to reserve a zenodo DOI when using the github webhook, so we would have to revert to manual submissions.

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst:13]: In my opinion, it does not look that ugly in GitHub but let's hear people's opinions.

Comment (ntessore) [reply CITATION.rst:13]: I find a `pull-quote` more apropos here, and it should render nicely in either form.

File: CITATION.rst
Comment (ntessore) [reply CITATION.rst:14]: I suppose the majority of astro-ph people will use whatever citation comes out of ADS, so we should recommend that or we will only increase confusion.

File: CITATION.rst
Comment (Lucia-Fonseca) [reply CITATION.rst:13]: Do you mean this?
```suggestion
  .. pull-quote::

  This research made use of SkyPy, a Python package for forward modeling
  astronomical surveys (Amara et. al., 2021, SkyPy Collaboration, 202x).
```

File: CITATION.rst
Comment (itrharrison) [reply CITATION.rst:14]: Both Planck and DES have managed to have "[X] Collaboration" as an author on ADS. It seems as though this is picked up from the author list on the arXiv submission. I have also had good experience of the "Feedback -> Submit Updates" option on ADS.

Once a DOI and ADS Bibcode is generated, it would hopefully be possible to update it in this way.

Comment (itrharrison) [reply CITATION.rst:14]: @rrjbca 
Screenshot 2021-09-13 at 15 42 06

Comment (Lucia-Fonseca) [reply CITATION.rst:14]: Is this something we want to try?
I'd rather have SkyPy Collaboration rather than Amara et al. 2021 but:
* Is it worth delaying the release?
* Sutieng's paper is ready and they cite Amara et al 2021.

Comment (rrjbca) [reply CITATION.rst:14]: In my opinion, releasing this PR is our single highest priority. The proposed guidelines are adequate and these side issues are unresolved and unnecessarily holding us up. Like it or not, Amara et al 2021 is the correct citation given by both ADS and JOSS. I agree that our citation guidelines should be consistent with ADS in the first instance. I'm also against modifying the ADS entry at the expense of consistency with the entry on the JOSS website. Similarly, reserving a DOI has not been tested with the GitHub webhook and putting the wrong DOI in our citation guidelines could be incredibly harmful. I would suggest trying to reserve a DOI for our next release and if we can demonstrate that it works then integrate reserved DOIs into the citation guidelines for future releases.

Comment (ntessore) [reply CITATION.rst:14]: I don't disagree necessarily with any of this, but I think it is not fair labeling this a "side issue" or "unnecessary". We agreed pretty early on that all publications would be authored by "SkyPy collaboration" with an alphabetical author list following. Now reneging on this merits at least a discussion, and probably a much more visible one than hidden in this PR somewhere.

Comment (rrjbca) [reply CITATION.rst:14]: Totally agree these are decisions beyond the scope of this PR and review that should be discussed by the collaboration. But the current proposal is at least accurate and waiting for the resolution of that discussion could cost us citations. The guidelines can always be updated in the future.

Comment (Lucia-Fonseca) [reply CITATION.rst:14]: Agreed, we'd need a conversation with the collaboration and we can also merge this as it is for the sake of the v0.5 release. After all, the issue is not about the author list (which I agree it's not a secondary issue but it is not a GitHub issue either)

File: CITATION.rst
Comment (rrjbca) [inline CITATION.rst:12]:
```suggestion
```
This adds nothing over a standard block quote without additional formatting

**Review by rrjbca — APPROVED** [review]
I'm happy to approve this PR in it's current state (although possibly check my brief comment r.e. unformatted pull-quote) but will wait for others involved to approve before merging.

**Comment by Lucia-Fonseca** [comment]
After offline conversations, we agree the citation would be Amara et al. 2021 and not the preferred option SkyPy Collaboration 2021 for justified reasons. Therefore, this PR can be merged.

**Review by rrjbca — APPROVED** [review]

**Pull request merged**
